### PR TITLE
 GitHub plugin add deployments

### LINKED ISSUE #5949: [Feature][Deployments] Collect `deployments` from GitLab/GitHub
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

As a tech lead, I use GitLab/GitHub deployments to do deployments, thus I want to measure DORA with these.

### Description

Now, DevLake only collects the `deployments` entity from Bamboo and BitBucket; while GitLab and GitHub's deployments are not collected. With the introduction of the independent `deployment` table in #5475, it's natural to collect these entities too.

After these entities are introduced, the configuration of `deployments` for GitLab and GitLab can also be simpler for users.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. collect github deployment in plugin with graphql API
2. extract and convert deployments
3. update some comments
4. set environment as PRODUCTION by regexp, need to update Config UI @mintsweet 
5. covnert data to `cicd_deployment_commits` also

### Does this close any open issues?
Closes #5949 

### Screenshots
Include any relevant screenshots here.
#### raw table
image
#### tool table
image
#### domain tables
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by klesh — COMMENTED** [review]
I suggest that we put all model definition and migration scripts into `github` plugin.

File: backend/plugins/github/models/migrationscripts/20230913_add_tool_demployments.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+package migrationscripts
+
+import (
+	"github.com/apache/incubator-devlake/core/context"
+	"github.com/apache/incubator-devlake/core/errors"
+	"github.com/apache/incubator-devlake/core/models/migrationscripts/archived"
+	"time"
+)
+
+type GithubDeployment struct {
```
Comment (abeizn) [inline backend/plugins/github/models/migrationscripts/20230913_add_tool_demployments.go:27]: If this is the first time to add a table, please put this struct in archived

**Comment by d4x1** [comment]
> I suggest that we put all model definition and migration scripts into `github` plugin.

Good advice, updated!

File: backend/plugins/github/models/migrationscripts/20230913_add_tool_demployments.go (hunk 1 above)
Comment (d4x1) [reply backend/plugins/github/models/migrationscripts/20230913_add_tool_demployments.go]: Got it.Updated!

**Comment by d4x1** [comment]
> I suggest that we put all model definition and migration scripts into `github` plugin.

done.

**Review by klesh — APPROVED** [review]

**Pull request merged**